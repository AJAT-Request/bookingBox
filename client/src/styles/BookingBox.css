:local(.container) {
  margin: 40px;
  width: 376px;
  border: 1px solid rgba(0, 0, 0, 0.15);
  background-color: white;
  color: rgb(72, 72, 72);
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
}

:local(.content) {
  margin: 20px;
}

:local(.line) {
  margin-top: 20px;
  margin-bottom: 20px;
  border: solid rgba(0, 0, 0, 0.15);
  border-width: 1px 0px 0px 0px;
}
