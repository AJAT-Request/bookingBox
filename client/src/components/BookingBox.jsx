import NightlyFee from './NightlyFee';
import Reviews from './Reviews';
import Dates from './Dates';
import Guests from './Guests';
import SumTotals from './SumTotals';
import BookButton from './BookButton';
import styles from '../styles/BookingBox.css';

const React = require('react');
const axios = require('axios');

class BookingBox extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      nightlyFee: null,
      serviceFee: null,
      cleaningFee: null,
      numReviews: null,
      reviewRating: null,
      numGuests: null,
      daysBooking: 0,
      render: false,
    };

    this.setNumDays = this.setNumDays.bind(this);
  }
  // (listingName, nightlyFee, serviceFee, cleaningFee, numReviews,
  // reviewRating, numGuests, timesRecentlyViewed)

  componentDidMount() {
    axios.get('/bookingBox/info', {
      params: {
        roomId: window.location.pathname.slice(7),
      },
    }).then((response) => {
      const result = response.data;
      this.setState({
        nightlyFee: result[0].nightlyFee,
        serviceFee: result[0].serviceFee,
        cleaningFee: result[0].cleaningFee,
        numReviews: result[0].numReviews,
        reviewRating: result[0].reviewRating,
        numGuests: result[0].numGuests,
      });
    }).catch((error) => {
      console.log('ERROR IN AXIOS GET: ', error);
    });
  }

  setNumDays(number) {
    // Use this function inside the calendar
    console.log('DAY CHANGED', number);
    this.setState({
      daysBooking: number,
    });
  }

  toggleTotals() {
    const { render } = this.state;
    this.setState({
      render: !render,
    });
  }

  render() {
    const {
      nightlyFee, serviceFee, cleaningFee, numReviews, reviewRating, numGuests, daysBooking, render,
    } = this.state;
    if (!nightlyFee) {
      return <div>Could not find room!</div>;
    }
    return (
      <div className={styles.container}>
        <div className={styles.content}>
          <NightlyFee price={nightlyFee} />
          <Reviews rating={reviewRating} num={numReviews} />
          <div className={styles.line} />
          <Dates setNumDays={this.setNumDays} />
          <div className={styles.line} />
          <Guests numGuests={numGuests} />
          <SumTotals
            nightlyFee={nightlyFee}
            serviceFee={serviceFee}
            cleaningFee={cleaningFee}
            days={daysBooking}
            render={render}
          />
          <BookButton />
          <div className={styles.line} />
          <button type="submit" onClick={this.toggleTotals.bind(this)}>Render Totals!</button>
        </div>
      </div>
    );
  }
}

export default BookingBox;
