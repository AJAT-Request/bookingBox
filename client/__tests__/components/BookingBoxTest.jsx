import React from 'react';
// import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import BookingBox from '../../src/components/BookingBox';

it('should add', () => {
  const two = 2;
  expect(1 + 1).toEqual(two);
});

describe('BookingBox', () => {
  it('renders correctly', () => {
    const component = shallow(
      <BookingBox />,
    );
    expect(component).toMatchSnapshot();
  });
});
