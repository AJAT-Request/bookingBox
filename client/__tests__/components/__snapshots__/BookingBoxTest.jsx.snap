// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingBox renders correctly 1`] = `
<div>
  Could not find room!
</div>
`;
